components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -100,44 +104,70 @@ func (backend ES2Backend) getNodeReportIdsFromTimeseries(esIndex string,
 			"took",
 			"hits.total",
 			"aggregations.nodes.buckets.key",
-			"aggregations.nodes.buckets.distinct.hits.hits._id").
+			"aggregations.nodes.buckets.distinct.hits.hits._id",
+			"aggregations.nodes.buckets.distinct.hits.hits._source",
+		).
 		SearchSource(searchSource).
 		Do(context.Background())
 
 	if err != nil {
 		logrus.Errorf("unable to getNodeReportIdsFromTimeseries %v", err)
-		return []string{}, errors.Wrap(err, "getNodeReportIdsFromTimeseries unable to complete search")
+		return &repIds, errors.Wrap(err, "getNodeReportIdsFromTimeseries unable to complete search")
 	}
 
 	if searchResult.TotalHits() == 0 || searchResult.Hits.TotalHits == 0 {
 		logrus.Debugf("getNodeReportIdsFromTimeseries: No report ids for the given filters: %+v\n", filters)
 		// no matching report IDs is not an error, just return an empty array
-		return []string{}, nil
+		return &repIds, nil
 	}
+	LogQueryPartMin(esIndex, searchResult.Aggregations, fmt.Sprintf("%s searchResult aggs", myName))
 
 	outermostAgg, _ := searchResult.Aggregations.Terms("nodes")
 	if outermostAgg != nil {
 		for _, nodeBucket := range outermostAgg.Buckets {
 			topHits, _ := nodeBucket.Aggregations.TopHits("distinct")
 			nodeID := fmt.Sprintf("%s", nodeBucket.Key)
 			for _, hit := range topHits.Hits.Hits {
-				nodeReport[nodeID] = hit.Id
+				var et EndTimeSource
+				if hit.Source != nil {
+					err = json.Unmarshal(*hit.Source, &et)
+					if err != nil {
+						logrus.Errorf("could not get the end_time for runid: %s %v", hit.Id, err)
+						continue
+					}
+				}
+				endTimeTimestamp, _ := ptypes.TimestampProto(et.EndTime)
```

Comment:
error check?

Reply:
I didn't do anything with this error because it marshaled without error in the line previously, so it's value won't be nil but, worst case, the zero value of go's "time.Time" which is the first moment in time (January 1, year 1, 00:00:00.000000000 UTC)... I figured that by continuing on here, we'd allow the user decide what to do with that.. for example, they could notice that in the case that the end_time is the zero value and then decide to look over a wide range, etc.. but at least they'd have the run_id.   I was torn about how and to handle this one.

Thanks for pointing it out.  I will print a log message when an error occurs here.